hack/observability/cluster-api-visualizer/fetch-visualizer-resources.sh

```diff
@@ -18,7 +18,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-CHART_RELEASE=${CHART_RELEASE:-1.0.0}
+CHART_RELEASE=${CHART_RELEASE:-1.0.2}
```

Comment:
If I run `fetch-visualizer-resources.sh`, it produces a diff. Was this a further change after running the script, because the `apiGroups` defined in the chart are too restrictive?

```diff
diff --git a/hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml b/hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml
index 190f11d9..c65d2ca1 100644
--- a/hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml
+++ b/hack/observability/cluster-api-visualizer/chart/templates/clusterrole.yaml
@@ -3,9 +3,30 @@ kind: ClusterRole
 metadata:
   name: capi-visualizer
 rules:
-# TODO: Consider restricting the rights to the relevant CAPI apiGroups.
 - apiGroups:
+  - ''
+  resources:
   - '*'
+  verbs:
+  - 'list'
+  - 'get'
+  - 'watch'
+- apiGroups:
+  - 'apiextensions.k8s.io'
+  resources:
+  - 'customresourcedefinitions'
+  verbs:
+  - 'list'
+  - 'get'
+  - 'watch'
+- apiGroups:
+  - 'cluster.x-k8s.io'
+  - 'addons.cluster.x-k8s.io'
+  - 'bootstrap.cluster.x-k8s.io'
+  - 'controlplane.cluster.x-k8s.io'
+  - 'ipam.cluster.x-k8s.io'
+  - 'infrastructure.cluster.x-k8s.io'
+  - 'runtime.cluster.x-k8s.io'
   resources:
   - '*'
   verbs:
```

Reply:
yes this is what i expected , the upstream chart change should be ( once vendored through the script ) 

* RBAC changes ( those in your diff )
* a change in the image ( which i don't see and i will double check now )

Reply:
the image is `latest` with `always pull` so that's why there is no diff for that ( https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/hack/observability/cluster-api-visualizer/chart/values.yaml )

so this looks good to me , we should have this RBAC changes vendored in here

Reply:
@Jont828 what is the outcome of this discussion, is the PR ready?

Reply:
we do need to run the script and add the updated clusterrole.yaml file to the PR to get the version `1.0.2` of the chart vendored in. 

I can do that, just did not want to mess with anyone else working on it :)

Reply:
Done, just ran the script!